libraries/libfc/include/fc/container/ordered_diff.hpp

```diff
@@ -26,10 +26,14 @@ namespace fc {
  * @param Container container type for ordered diff and for diff_result
  */
 template <typename T, typename SizeType = size_t, template<typename Y, typename...> typename Container = std::vector>
-requires std::equality_comparable<T> && std::random_access_iterator<typename Container<T>::iterator>
+requires std::equality_comparable<T>
+         && std::random_access_iterator<typename Container<T>::iterator>
+         && std::is_unsigned_v<SizeType>
```

Comment:
Since we use `Container<T>::size_type` as type of the index of our loops, let's require that it is an unsigned type as well:
`&& std::is_unsigned_v<typename Container<T>::size_type>`

This will ensure we don't compare signed and unsigned values.